### PR TITLE
Cleanup some old python 2 idioms

### PR DESCRIPTION
**Pull request by neutrinoceros**
- Inheritance from `object` class is implicit in python 3 so there's no need to keep enforcing it after we dropped support for python 2.

The following were done semi-automatically through `2to3` + manual validation, since in many cases this tool will produce incorrect results (e.g. with sorting numpy arrays)
- cleanup metaclass declarations
- remove future imports
- cleanup rare reliquary occurrences of `long` (-> `int`) and `unicode` (-> `str`)
- ⚠️ two classes were implementing a `next` method instead of `__next__`, this change could break some downstream code if the deprecated syntax `self.next()` was used (should be a niche case and is easily fixed)
- use builtin `a = sorted(data)` instead of `a=list(data); a.sort()` where relevant (i.e. where it doesn't cause performance hits by adding a copy step)
- consistently use `while True` instead of `while 1`
- cleanup type comparisons: use builtin `isinstance(a, mytype)` instead of `type(a) is mytype`.  Within tests I conservatively assumed it's preferable to leave the more strict type comparison but replaced some `==` comparisons with `is` (I don't know if it's even possible to build case where they are not equivalent but the later is definetely correct).

### CODE REVIEW

File: yt/frontends/swift/tests/test_outputs.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 @requires_file(keplerian_ring)
 def test_non_cosmo_dataset():
     ds = load(keplerian_ring)
-    assert(type(ds) == SwiftDataset)
+    assert(type(ds) is SwiftDataset)
```
Comment (cphyc) [inline yt/frontends/swift/tests/test_outputs.py:18]:
```suggestion
    assert(isinstance(ds, SwiftDataset)
```

File: yt/geometry/tests/fake_octree.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-from __future__ import print_function
+
 from yt.geometry.fake_octree import create_fake_octree
```
Comment (cphyc) [inline yt/geometry/tests/fake_octree.py:2]:
```suggestion
from yt.geometry.fake_octree import create_fake_octree
```

File: yt/mods.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-from __future__ import absolute_import
+
 
 #
```
Comment (cphyc) [inline yt/mods.py:3]:
```suggestion
#
```

File: yt/utilities/decompose.py [hunk 4]
```diff
@@ -1,4 +1,4 @@
-from __future__ import division
+
 
 import numpy as np
```
Comment (cphyc) [inline yt/utilities/decompose.py:3]:
```suggestion
import numpy as np
```

File: yt/utilities/grid_data_format/__init__.py [hunk 5]
```diff
@@ -1,3 +1,3 @@
-from __future__ import absolute_import
+
 from .conversion import *
```
Comment (cphyc) [inline yt/utilities/grid_data_format/__init__.py:2]:
```suggestion
from .conversion import *
```

File: yt/utilities/grid_data_format/conversion/__init__.py [hunk 6]
```diff
@@ -1,4 +1,4 @@
-from __future__ import absolute_import
+
 from .conversion_abc import Converter
```
Comment (cphyc) [inline yt/utilities/grid_data_format/conversion/__init__.py:2]:
```suggestion
from .conversion_abc import Converter
```

File: yt/utilities/grid_data_format/conversion/conversion_athena.py [hunk 7]
```diff
@@ -1,5 +1,5 @@
-from __future__ import print_function
-from __future__ import absolute_import
+
+
 import numpy as np
```
Comment (cphyc) [inline yt/utilities/grid_data_format/conversion/conversion_athena.py:3]:
```suggestion
import numpy as np
```

File: yt/utilities/lib/cykdtree/tests/test_utils.py [hunk 8]
```diff
@@ -1,4 +1,4 @@
-from __future__ import division
+
 
 import numpy as np
```
Comment (cphyc) [inline yt/utilities/lib/cykdtree/tests/test_utils.py:3]:
```suggestion
import numpy as np
```

File: yt/utilities/sdf.py [hunk 9]
```diff
@@ -1,4 +1,4 @@
-from __future__ import print_function
+
 from io import StringIO
```
Comment (cphyc) [inline yt/utilities/sdf.py:2]:
```suggestion
from io import StringIO
```

File: yt/visualization/volume_rendering/lens.py [hunk 10]
```diff
@@ -1,4 +1,4 @@
-from __future__ import division
+
 from yt.funcs import mylog
```
Comment (cphyc) [inline yt/visualization/volume_rendering/lens.py:2]:
```suggestion
from yt.funcs import mylog
```

File: yt/data_objects/tests/test_octree.py [hunk 11]
```diff
@@ -12,7 +12,7 @@ def test_building_tree():
     '''
     ds = fake_sph_grid_ds()
     octree = ds.octree(n_ref=1)
-    assert(type(octree) == YTOctree)
+    assert(type(octree) is YTOctree)
```
[the remark this answers is not in the document]
Comment (neutrinoceros) [reply yt/data_objects/tests/test_octree.py:15]: In the context of a test that I didn't write I assume it's best to check for exact typing. Using `isinstance()` here would make the test more flexible and I see no reason to allow this as long as it's not motivated by the data we actually test.

File: yt/frontends/swift/tests/test_outputs.py (hunk 1 above)
Comment (neutrinoceros) [reply yt/frontends/swift/tests/test_outputs.py:18]: same as above, not willing to make tests more flexible here.

**Comment by neutrinoceros** [comment]
Oops. I used `[ci-skip]` instead of `[ci skip]` and now the CI pipeline will be running for ever 🤦

File: yt/geometry/coordinates/tests/test_cartesian_coordinates.py [hunk 12]
```diff
@@ -13,8 +13,7 @@ def test_cartesian_coordinates():
     # We're going to load up a simple AMR grid and check its volume
     # calculations and path length calculations.
     ds = fake_amr_ds()
-    axes = list(set(ds.coordinates.axis_name.values()))
-    axes.sort()
+    axes = sorted(set(ds.coordinates.axis_name.values()))
```
Comment (munkm) [inline yt/geometry/coordinates/tests/test_cartesian_coordinates.py:16]: 👍 to this change

File: yt/utilities/math_utils.py [hunk 13]
```diff
@@ -819,31 +819,30 @@ def quartiles(a, axis=None, out=None, overwrite_input=False):
     """
     if overwrite_input:
         if axis is None:
-            sorted = a.ravel()
-            sorted.sort()
+            a_sorted = sorted(a.ravel())
         else:
             a.sort(axis=axis)
-            sorted = a
+            a_sorted = a
```
Comment (munkm) [inline yt/utilities/math_utils.py:825]: I like this clearer variable name

File: yt/geometry/coordinates/spherical_coordinates.py [hunk 14]
```diff
@@ -1,4 +1,4 @@
-from __future__ import print_function
+
```
[the remark this answers is not in the document]
Comment (neutrinoceros) [reply yt/geometry/coordinates/spherical_coordinates.py:1]: good catch, fixing it right now

**Comment by neutrinoceros** [comment]
@yt-fido test this please

**Comment by neutrinoceros** [comment]
@yt-fido test this please

**Pull request merged**